eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/GoodLoggingCheck.java

```diff
@@ -0,0 +1,181 @@
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Good Logging Practice:
+ * <ol>
+ * <li>Non-static Logger instance.</li>
+ * <li>Logger variable name should be 'logger' in public class and not exist in an implementation package.</li>
+ * <li>Should not use any external logger class, only use ClientLogger. No slf4j, log4j, or other logging imports are allowed.</li>
+ * <li>'System.out' and 'System.err' is not allowed as well.</li>
+ * </ol>
+ */
+public class GoodLoggingCheck extends AbstractCheck {
+    private static final String CLIENT_LOGGER_PATH = "com.azure.core.util.logging.ClientLogger";
+    private static final String CLIENT_LOGGER = "ClientLogger";
+    private static final String LOGGER_NAME_ERROR = "ClientLogger instance naming: use ''%s'' instead of ''%s'' for consistency.";
+    private static final String STATIC_LOGGER_ERROR = "Reference to ClientLogger should not be static: remove static modifier.";
+    private static final String NOT_CLIENT_LOGGER_ERROR = "Do not use external logger class. Use ''%s'' as a logging mechanism instead of ''%s''.";
+
+    private String invalidLogger;
+    private boolean hasClientLoggerImported;
+    private String loggerRequiredName;
+    private String className;
+
+    private static final Set<String> INVALID_LOG_SET = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(
+        "org.slf4j", "org.apache.logging.log4j"
+    )));
+
+    /**
+     * Setter to specifies valid identifiers
+     * @param loggerRequiredName the variable name of logger used
+     */
+    public void setLoggerName(String loggerRequiredName) {
+        this.loggerRequiredName = loggerRequiredName;
+    }
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {
+            TokenTypes.IMPORT,
+            TokenTypes.LITERAL_CLASS,
+            TokenTypes.LITERAL_NEW,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.METHOD_CALL
+        };
+    }
+
+    @Override
+    public void finishTree(DetailAST ast) {
+        invalidLogger = null;
+        hasClientLoggerImported = false;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        switch (ast.getType()) {
+            case TokenTypes.IMPORT:
+                String importClassPath = FullIdent.createFullIdentBelow(ast).getText();
+                hasClientLoggerImported = hasClientLoggerImported || importClassPath.equals(CLIENT_LOGGER_PATH);
+                for (final String logger : INVALID_LOG_SET) {
+                    if (importClassPath.startsWith(logger)) {
+                        invalidLogger = logger;
```

Comment:
You can log the error here directly instead of keeping track of the import and then logging the error in `LITERAL_CLASS`. Importing a different log package is sufficient to flag the issue.